cdap-ui/app/cdap/components/PipelineScheduler/BasicView/StartingAt.js

```diff
@@ -0,0 +1,158 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import SelectWithOptions from 'components/SelectWithOptions';
+import {MINUTE_OPTIONS, HOUR_OPTIONS_CLOCK, AM_PM_OPTIONS, ACTIONS as PipelineSchedulerActions} from 'components/PipelineScheduler/Store';
+import {updateCron} from 'components/PipelineScheduler/Store/ActionCreator';
+import {connect} from 'react-redux';
+
+const mapStateToStartingAtMinuteProps = (state) => {
+  return {
+    options: MINUTE_OPTIONS,
+    value: state.startingAtMinute
+  };
+};
+const mapDispatchToStartingAtMinuteProps = (dispatch) => {
+  return {
+    onChange: (e) => {
+      dispatch({
+        type: PipelineSchedulerActions.SET_STARTING_AT_MINUTE,
+        payload: {
+          startingAtMinute: e.target.value
+        }
+      });
+      updateCron();
+    }
+  };
+};
+
+const mapStateToStartingAtHourProps = (state) => {
+  return {
+    options: HOUR_OPTIONS_CLOCK,
+    value: state.startingAtHour
+  };
+};
+const mapDispatchToStartingAtHourProps = (dispatch) => {
+  return {
+    onChange: (e) => {
+      dispatch({
+        type: PipelineSchedulerActions.SET_STARTING_AT_HOUR,
+        payload: {
+          startingAtHour: e.target.value
+        }
+      });
+      updateCron();
+    }
+  };
+};
+
+const mapStateToStartingAtAMPMProps = (state) => {
+  return {
+    options: AM_PM_OPTIONS,
+    value: state.startingAtAMPM
+  };
+};
+const mapDispatchToStartingAtAMPMProps = (dispatch) => {
+  return {
+    onChange: (e) => {
+      dispatch({
+        type: PipelineSchedulerActions.SET_STARTING_AT_AM_PM,
+        payload: {
+          startingAtAMPM: e.target.value
+        }
+      });
+      updateCron();
+    }
+  };
+};
+
+const SelectStartingAtMinute = connect(
+  mapStateToStartingAtMinuteProps,
+  mapDispatchToStartingAtMinuteProps
+)(SelectWithOptions);
+const SelectStartingAtHour = connect(
+  mapStateToStartingAtHourProps,
+  mapDispatchToStartingAtHourProps
+)(SelectWithOptions);
+const SelectStartingAtAMPM = connect(
+  mapStateToStartingAtAMPMProps,
+  mapDispatchToStartingAtAMPMProps
+)(SelectWithOptions);
+
+const mapStateToStartingAtComponentProps = (state) => {
+  return {
+    intervalOption: state.intervalOption
+  };
+};
+
+const StartingAtComponent = ({intervalOption}) => {
+  let SelectComponent;
+
+  switch (intervalOption) {
+    case 'Hourly':
```

Comment:
Shouldn't this be `case INTERVAL_OPTIONS['HOURLY']` like the `SummaryComponent`?